src/Core/Addon/Theme/ThemeManagerBuilder.php

```diff
@@ -30,31 +30,44 @@
 use Db;
 use PrestaShop\PrestaShop\Adapter\Configuration;
 use PrestaShop\PrestaShop\Adapter\Hook\HookInformationProvider;
-use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
 use PrestaShop\PrestaShop\Core\Image\ImageTypeRepository;
 use PrestaShop\PrestaShop\Core\Module\HookConfigurator;
 use PrestaShop\PrestaShop\Core\Module\HookRepository;
+use PrestaShopBundle\Service\TranslationService;
+use PrestaShopBundle\Translation\Provider\Factory\ProviderFactory;
 use Shop;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Finder\Finder;
 
 class ThemeManagerBuilder
 {
     private $context;
     private $db;
+    /**
+     * @var ThemeValidator|null
+     */
     private $themeValidator;
     /**
-     * @var ContainerInterface|null
+     * @var TranslationService
      */
-    private $container;
+    private $translationService;
+    /**
+     * @var ProviderFactory
+     */
+    private $providerFactory;
 
-    public function __construct(Context $context, Db $db, ThemeValidator $themeValidator = null, ?ContainerInterface $container = null)
-    {
+    public function __construct(
+        Context $context,
+        Db $db,
+        TranslationService $translationService,
+        ProviderFactory $providerFactory,
+        ThemeValidator $themeValidator = null
+    ) {
         $this->context = $context;
         $this->db = $db;
         $this->themeValidator = $themeValidator;
-        $this->container = $container ?? SymfonyContainer::getInstance();
+        $this->translationService = $translationService;
+        $this->providerFactory = $providerFactory;
```

Comment:
I think you can still use `SymfonyContainer::getInstance()` to get `translationService` & `providerFactory` in case they are null. It would help the code being more DRY. WDYT?